bika/health/profiles/default/workflows/senaite_health_doctor_workflow/definition.xml

```diff
@@ -7,104 +7,150 @@
-
- <variable variable_id="actor" for_catalog="False" for_status="True" update_always="True">
-  <description i18n:attributes="title">The ID of the user who performed the previous transition</description>
-  <default>
-   <expression>user/getId</expression>
-  </default>
-  <guard>
-  </guard>
- </variable>
-
- <variable variable_id="comments" for_catalog="False" for_status="True" update_always="True">
-  <description i18n:attributes="title">Comment about the last transition</description>
-  <default>
-   <expression>python:state_change.kwargs.get('comment', '')</expression>
-  </default>
-  <guard>
-  </guard>
- </variable>
-
- <variable variable_id="review_history" for_catalog="False" for_status="False" update_always="False">
-  <description i18n:attributes="title">Provides access to workflow history</description>
-  <default>
-   <expression>state_change/getHistory</expression>
-  </default>
-  <guard>
-   <guard-permission>Request review</guard-permission>
-   <guard-permission>Review portal content</guard-permission>
-  </guard>
- </variable>
-
- <variable variable_id="time" for_catalog="False" for_status="True" update_always="True">
-  <description i18n:attributes="title">When the previous transition was performed</description>
-  <default>
-   <expression>state_change/getDateTime</expression>
-  </default>
-  <guard>
-  </guard>
- </variable>
+  <permission>Access contents information</permission>
+  <permission>Delete objects</permission>
+  <permission>Modify portal content</permission>
+  <permission>View</permission>
+
+  <!-- Senaite permissions for transitions (govern guards) -->
+  <permission>senaite.core: Transition: Activate Doctor</permission>
+  <permission>senaite.core: Transition: Deactivate Doctor</permission>
+
+  <!-- State: active -->
+  <state state_id="active" title="Active" i18n:attributes="title">
+    <exit-transition transition_id="deactivate" />
+
+    <!-- Transition permissions -->
+    <permission-map name="senaite.health: Transition: Deactivate Doctor" acquired="True" />
+
+   <permission-map name="Access contents information" acquired="False">
+    <permission-role>Manager</permission-role>
+    <permission-role>LabManager</permission-role>
+    <permission-role>LabClerk</permission-role>
+    <permission-role>Analyst</permission-role>
+    <permission-role>Sampler</permission-role>
+    <permission-role>Preserver</permission-role>
+    <permission-role>Owner</permission-role>
+    <permission-role>Doctor</permission-role>
+    <permission-role>Client</permission-role>
+   </permission-map>
+   <permission-map name="Delete objects" acquired="False"/>
+   <permission-map name="Modify portal content" acquired="False">
+    <permission-role>Manager</permission-role>
+    <permission-role>LabManager</permission-role>
+    <permission-role>LabClerk</permission-role>
+    <permission-role>Analyst</permission-role>
+    <permission-role>Sampler</permission-role>
+    <permission-role>Preserver</permission-role>
```

Comment:
I would not allow Analysts, Samplers and Preservers to be able to modify a Doctor